lib/angular/directive.js

```diff
@@ -38,4 +41,31 @@ angular.module('wfm.file.directives', [])
     }
   };
 })
+  .directive('fileDetail', function($templateCache) {
+    /*
+     detail view of a file, passed in is file object and options.
+     Options help to chose which details we want to render:
+     e.g. <file-detail file="ctrl.file" options="id, name, uid, owner, preview">
+     would render all details.
+     Options should be comma separated ', '
+     */
+    return {
+      restrict: 'E',
+      template: $templateCache.get('wfm-template/file-detail.tpl.html')
+      , scope: {
+        file: '=',
+        options: '@'
+      }
+      , controller: function($scope) {
+        var self = this;
+        self.file = $scope.file;
+        self.options = {};
+        var opts = $scope.options ? $scope.options.split(', ') : [];
```

Comment:
I think you could just do the same as is done for the $scope.file

```
<file-detail file="ctrl.file" options="ctrl.fields">

</file-detail>
```

and assign an array of fields to the controller scope in the relevant app. This would save you from having to do the split etc. It would just be an array.